File: go/example_code/cloudtrail/describe_trails.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"fmt"

	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/cloudtrail"
)

func main() {
	// Initialize a session that the SDK will use to load
	// credentials from the shared credentials file ~/.aws/credentials.
	sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}))

	// Create CloudTrail client
	svc := cloudtrail.New(sess)

	resp, err := svc.DescribeTrails(&cloudtrail.DescribeTrailsInput{TrailNameList: nil})
	if err != nil {
		fmt.Println("Got error calling CreateTrail:")
		fmt.Println(err.Error())
		return
	}

	fmt.Println("Found", len(resp.TrailList), "trail(s)")
	fmt.Println("")

	for _, trail := range resp.TrailList {
		fmt.Println("Trail name:  " + *trail.Name)
		fmt.Println("Bucket name: " + *trail.S3BucketName)
		fmt.Println("")
	}
}

File: ruby/example_code/cloudtrail/describe_trails.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# Demonstrates how to retrieve details about AWS CloudTrail trails

# snippet-start:[cloudtrail.Ruby.describeTrails]
require "aws-sdk-cloudtrail"  # v2: require 'aws-sdk'

def describe_trails_example(client)
  resp = client.describe_trails({})
  puts "Found #{resp.trail_list.count} trail(s)."

  resp.trail_list.each do |trail|
    puts "Name:           " + trail.name
    puts "S3 bucket name: " + trail.s3_bucket_name
    puts
  end
  # snippet-end:[cloudtrail.Ruby.describeTrails]
end

if __FILE__ == $0
  client = Aws::CloudTrail::Client.new
  describe_trails_example(client)
end
